src/lib/candidate.ts

```diff
@@ -0,0 +1,55 @@
+#!/usr/bin/env node
+import { baseGitHubWorkflow } from "~common/github-base-workflow";
+import type { Repo } from "~common/repo";
+import { FilterMatchHeuristic, calcHeuristics } from "~filters";
+import { generateWorkflowSteps } from "~generators";
+
+export type WorkflowFile = {
+  name: string,
+  workflow: object
+}
+
+export const validateRepo = (name: string): boolean => {
+  // Evaluates if a repo is a candidate or not
+  return false;
+}
+
+export const genWorkflows = async (repo: Repo): Promise<WorkflowFile[]> => {
+  let toolNames = await findBuildTools(repo);
+  let workflows = [];
+
+  toolNames.forEach(toolName => {
```

Comment:
this can be a map and then you can do a 

```
return { name: `${toolName}-deploy`, workflow }
```
at the end of the function

so this line would look something like: 
`return toolNames.map(toolName => {`